superduperdb/ext/llm/model.py

```diff
@@ -0,0 +1,398 @@
+import dataclasses as dc
+import functools
+import os
+import typing
+import typing as t
+
+import torch
+import transformers
+from transformers import (
+    AutoModelForCausalLM,
+    AutoTokenizer,
+    BitsAndBytesConfig,
+    TrainingArguments,
+)
+from typing_extensions import Optional
+
+from superduperdb import logging
+from superduperdb.backends.query_dataset import query_dataset_factory
+from superduperdb.components.model import (
+    Model,
+    _TrainingConfiguration,
+)
+from superduperdb.ext.utils import ensure_initialized
+
+if typing.TYPE_CHECKING:
+    from superduperdb.backends.base.query import Select
+    from superduperdb.base.datalayer import Datalayer
+    from superduperdb.components.dataset import Dataset
+    from superduperdb.components.metric import Metric
+
+
+@dc.dataclass
+class LLMTrainingArguments(TrainingArguments):
```

Comment:
Are we sure we want to hard code the arguments like this? Maybe it should be more flexible?

Reply:
Ok I see that it also inherits argument from `transformers` arguments.